parser/match_operations.go

```diff
@@ -437,16 +437,33 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
+
+		match0Ops := matches[amountMatch[0]].Operations
+		match1Ops := matches[amountMatch[1]].Operations
+
+		if len(match0Ops) == 0 || len(match1Ops) == 0 {
+			return errors.New("opposite amounts comparison error: no matched operations")
+		}
+		// check if all operations within the same match slice have the same amount
+		if len(match0Ops) > 1 {
+			if err := equalAmounts(match0Ops); err != nil {
+				return fmt.Errorf(
+					"%w: opposite amounts comparison error: amounts not equal within the 0th match operations", err)
```

Comment:
nit: I would use %d here and insert the value `amountMatch[0]` (not guaranteed to be the `0th`